docs(vault): add Vault worker filter documentation (#2404)
hashicorp/boundary: website/content/docs/concepts/filtering/worker-tags.mdx

  know that you have only one value, an equivalent would be `"/tags/region/0" == "us-east-1"`.

- Grouping: `("us-east-1" in "/tags/region" and "/name" == "web-prod-us-east-1") or "webservers" in "/tags/type"`

# Target Worker Filtering

> Once workers have tags, it is possible to use these tags to control which

Reviewer: ```suggestion
Once workers have tags, these tags may be used to control which
```

> workers are allowed to handle a given session by specifying a `worker_filter`

Reviewer: ```suggestion
manage a given session by specifying a `worker_filter`
```

Reviewer: Feel free to modify my suggestions above if it changes the intent of that sentence.

attribute when configuring [targets](/docs/concepts/domain-model/targets).

# Vault Worker Filtering <sup>HCP Only</sup>

> Tags can be used to control which [PKI workers] can handle Vault requests by specifying

Reviewer: ```suggestion
Tags can be used to control which [PKI workers] can manage Vault requests by specifying
```

> Tags can be used to control which [PKI workers] can handle Vault requests by specifying

Reviewer: ```suggestion
Tags are used to control which [PKI workers] can manage Vault requests by specifying
```

a `worker_filter`attribute when configuring [credential stores].

> This allows for the use of private Vault instances with Boundary. PKI workers

Reviewer: ```suggestion
This allows the use of private Vault instances with Boundary. PKI workers
```

deployed in the same network as a private Vault instance can access and relay
Vault requests to Boundary controllers.

 ~> **Note:** A PKI worker that matches the worker filter must exist before defining the Vault credential store, as it

>  will perform the Vault calls needed to set up the credential store with Boundary

Reviewer: ```suggestion
 will perform the Vault calls needed to set up the credential store with Boundary.
```
